workflow/scripts/osemosys_global/powerplant/main.py

```diff
@@ -0,0 +1,299 @@
+import pandas as pd
+import os
+
+from read import(
+    import_plexos_2015,
+    import_weo_regions,
+    import_weo_costs,
+    import_op_life,
+    import_naming_convention_tech,
+    import_afs,
+    import_custom_res_cap,
+)
+
+from constants import(
+    DUPLICATE_TECHS,
+    MODE_LIST,
+    RENEWABLES_LIST,
+    DF_IAR_CUSTOM_VAL,
+    DF_OAR_CUSTOM_VAL
+    )
+
+from data import(
+    set_generator_table,
+    average_efficiency,
+    )
+
+from residual_capacity import(
+    res_capacity,
+    add_custom_res_cap
+    )
+
+from activity import(
+    activity_master_start,
+    activity_output_pwr,
+    activity_input_pwr,
+    activity_upstream,
+    activity_master_end,
+    capact
+    )
+
+from costs import(
+    costs_pwr,
+    costs_end
+    )
+
+from operational_life import set_op_life
+
+from investment_constraints import cap_investment_constraints
+
+from sets import(
+    create_sets,
+    output_sets
+    )
+
+from user_defined_capacity import set_user_defined_capacity
+
+from availability import availability_factor
+
+def main(
+    plexos_prop: pd.DataFrame,
+    plexos_memb: pd.DataFrame,
+    weo_costs: pd.DataFrame,
+    weo_regions: pd.DataFrame,
+    default_op_life: pd.DataFrame,
+    naming_convention_tech: pd.DataFrame,
+    custom_res_cap: pd.DataFrame,
+    default_av_factors: pd.DataFrame,
+):
```

Comment:
`default_op_life`, `naming_converntion_tech`, and `default_av_factors` are not used at all here? If that is correct, can we remove them as inputs, the accompanying import data function calls, and the files required in the snakemake rule?

Reply:
Correct, they can be removed (forgot to that in my rush to get out of the door on Thursday) and have done that now.

Reply:
`default_op_life` and `naming_converntion_tech` were linked to the wrong variables (`op_life_dict` and `tech_code_dict` vs `op_life` and 'tech_list') but should be used and `default_av_factors` was missing in the call (`availability` was used directly). Amended accordingly.